### ISSUE TITLE
Generate a Doxygen TAG file

### ISSUE THREAD

**Issue by claudiofantacci** [robotology/how-to-export-cpp-library#40]

**Comment by claudiofantacci** [comment]
From #24.

**Comment by diegoferigo** [comment]
What do you mean with TAG, to handle the project version?

**Comment by claudiofantacci** [comment]
> What do you mean with TAG, to handle the project version?

cc @traversaro

**Comment by traversaro** [comment]
See the [`Doxytag`](https://www.star.bnl.gov/public/comp/sofi/doxygen/doxytag_usage.html) docs for more info. Basically they are useful to ensure cross-projects links in Doxygen documentation.

**Comment by diegoferigo** [comment]
I remember now that we discussed on this super cool feature long time ago. Since all our documentation is based on doxygen it would be great having a simple example running.

**Comment by claudiofantacci** [comment]
Do we have some sample code about this?

**Comment by traversaro** [comment]
See usage of `YARP_DOXYGEN_TAGFILE` and `DOX_GENERATE_TAGFILE` in https://github.com/robotology/yarp/blob/master/cmake/YarpDoc.cmake#L43 .

**Comment by traversaro** [comment]
Not sure if they do something similar in ignition-cmake, but I just saw that they have this supercool file that I guess make sure that std doxygen objects are linked to cppreference.com : https://bitbucket.org/ignitionrobotics/ign-cmake/src/default/doc/doxygen/cppreference.tag.xml .

**Comment by traversaro** [comment]
See also https://en.cppreference.com/w/Cppreference:Archives#Doxygen_tag_file .

**Comment by claudiofantacci** [comment]
Well, there is some work here to do. I reckon if we can make a standard template project for this it will be really super useful!

**Comment by traversaro** [comment]
Relevant ign-cmake PR: https://bitbucket.org/ignitionrobotics/ign-cmake/pull-requests/65/doxygen-macro-for-downstream-packages-30/diff .

**Comment by traversaro** [comment]
> Well, there is some work here to do. I reckon if we can make a standard template project for this it will be really super useful!

Definitely. Another tricky part is how to handle discovering and handling of TAG files of upstream libraries. This is something that the CMake config file could expose, but I am not aware of any C++ project that does this (that's the reason I wanted to check if they di something similar in the ignition projects). 

Once we have something like that in place, we can generate in one shot all the doxygen docs for the packages in a superbuild, in a way that they cross link between one and another.
